src/osdp_cp.c

```diff
@@ -1433,10 +1429,10 @@ static struct osdp *__cp_setup(int num_pd, const osdp_pd_info_t *info_list)
 		} else if (is_enforce_secure(pd)) {
 			LOG_PRINT("SCBK must be passed for each PD when"
 				  " ENFORCE_SECURE is requested.");
-			goto error;
+			return -1;
```

Comment:
Missing cleanup -- here and below? Memory allocated above is left dangling but the bigger concern is you have partial changes above.

Reply:
done.